src/v/redpanda/rp_services.h

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright 2023 Redpanda Data, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#include "utils/string_switch.h"
+
+#include <type_traits>
+
+enum class service_kind {
+    schema_registry,
+};
+
+constexpr std::string_view to_string_view(service_kind kind) {
+    switch (kind) {
+    case service_kind::schema_registry:
+        return "schema-registry";
+    }
+    return "invalid";
+}
+
+template<typename E>
+std::enable_if_t<std::is_enum_v<E>, std::optional<E>>
+  from_string_view(std::string_view);
+
+template<>
+constexpr std::optional<service_kind>
+from_string_view<service_kind>(std::string_view sv) {
+    return string_switch<std::optional<service_kind>>(sv)
+      .match(
+        to_string_view(service_kind::schema_registry),
+        service_kind::schema_registry)
+      .default_match(std::nullopt);
+}
```

Comment:
I wonder if this file should exist as part of the admin api. I'm not too keen on these being in global scope.

Reply:
Moving them to `admin_server.h` would make sense. I'll add that as a follow-up

Reply:
Done. See https://github.com/redpanda-data/redpanda/pull/8127/commits/41abb0255015a9d2350db7be17dd63731a9177ff